WhoIsThatServer.Storage/Controllers/TargetElementController.cs

```diff
@@ -42,5 +42,31 @@ public IHttpActionResult Post([FromBody] TargetElement targetElement)
 
             return Json(targetElement);
         }
+
+        /// <inheritdoc/>
+        [HttpPost]
+        [Route("api/game/{id}")]
+        public IHttpActionResult AssignRandomTarget(int id)
+        {
+            try
+            {
+                return Json(TargetElementHelper.AssignRandomTarget(id));
+            }
+
+            catch (ManagerException targetExistsException) when (targetExistsException.ErrorCode == StorageErrorMessages.TargetAlreadyAssignedError)
+            {
+                return Json(StorageErrorMessages.TargetAlreadyAssignedError);
+            }
+
+            catch (ManagerException noPlayersException) when (noPlayersException.ErrorCode == StorageErrorMessages.ThereAreNoPlayersError)
+            {
+                return Json(StorageErrorMessages.ThereAreNoPlayersError);
+            }
+
+            catch (ManagerException targetNotAssignedException) when (targetNotAssignedException.ErrorCode == StorageErrorMessages.TargetNotAssignedError)
+            {
+                return Json(StorageErrorMessages.TargetNotAssignedError);
```

Comment:
Damn, thats looking good! :) one thing. How does the consumer know that there was an error?

Reply:
Thanks!! Well, I gotta figure it out, but I thought that I could map this error message to frontend and display message there

Reply:
Well, even though it looks cleaner, you are still pretty much left with the same problem at the frontend/consumer. You need to return different status code (e.g 400), to know that you have failed, if the response in both success and failure cases are strings

Reply:
I still have a lot to learn, it seems :D 

But couldn't I just check which string I got from backend in frontend using mapped constants or something?

Reply:
Not exactly. And strings are especially dangerous. How do you know that I didnt enter "No face detected" as my name? Of course it's a silly example, but when you are asking for name, you should expect all kinds of inputs. So you should definitely know when you have failed by checking status at frontend

Reply:
So when you get a not success status, you know you have failed, whatever the error message you got. And THEN map error message to something, if needed

Reply:
So your controller should look something like this
```
return Ok(name);
...
return BadRequest(ErrorCode.SomethingHappened);
return BadRequest(ErrorCode.Error2);
```

Reply:
Fair point, I will look up on how to return bad status codes, as for now I have no idea. But yeah, I understand your point completely, I just need more time to work on it :D

Reply:
Well, I have provided you an answer already :DD this is the actual way to do that in .NET Core